apptainer/moose-dev.def

```diff
@@ -156,9 +140,47 @@ Fingerprints: 0CFFCAB55E806363601C442D211817B01E0911DB
     # Needed for coverage
     pip3 --no-cache install lcov-cobertura
 
-    # Build and install wasp
-    # This is redundant; hopefully we can use the one from conda in the future
-    WASP_SRC_DIR=${WASP_SRC_DIR} ${WASP_BUILD_SCRIPT} -D CMAKE_INSTALL_PREFIX:STRING=${WASP_DIR}
+{%- if WITH_LIBTORCH %}
+    # Libtorch-related vars
+    LIBTORCH_VER=2.3.0
+    LIBTORCH_DISTRIBUTION={{ WITH_LIBTORCH }}
+
+    # Clone pytorch
+    cd ${ROOT_BUILD_DIR}
+    git clone -b v${LIBTORCH_VER} --recursive https://github.com/pytorch/pytorch
+    cd pytorch
+
+    # Install the python requirements
+    pip install -r requirements.txt
+
+    # Below a certain version we need to downgrade our numpy
+    if { echo ${LIBTORCH_VER}; echo "2.2.0"; } | sort --version-sort --check=quiet; then
+      conda install -yq "numpy<2.0"
+    fi
+
+    # Fetch the patch that enables the successful compilation on this linux distro
+    curl -L https://github.com/pytorch/pytorch/pull/99468.patch > 99468.patch
+    git apply 99468.patch
+
+    # We have to diasble some errors considering they emerge as some optimization of the compiler on certain distributuins (sadly on this one to)
+    export CFLAGS+=" -Wno-error=maybe-uninitialized -Wno-error=uninitialized -Wno-error=restrict"
+    export CXXFLAGS+=" -Wno-error=maybe-uninitialized -Wno-error=uninitialized -Wno-error=restrict"
+
+    # Build and install
+    export CMAKE_PREFIX_PATH="${CONDA_PREFIX:-'$(dirname $(which conda))/../'}:${CMAKE_PREFIX_PATH}"
```

Comment:
Installing something like this directly into CONDA_PREFIX isn't great, because conda doesn't know about it. Is this what they suggest you do?

Reply:
Yeah these are the official instructions. So you are suggesting installing this somewhere else and adding it to python path?

Reply:
I don't think `CMAKE_PREFIX_PATH` has anything to do with "where to install the compiled libtorch libraries". Instead, it is simply used as a hint at CMake configure time.

Reply:
This PR will be closed, Logan took over this effort. The conclusion was also that the user will not get pytorch from the container, we will only build the libtorch part.